curl: Fix librtmp variant

tobbez opened the pull request:
Fix the `librtmp` variant by packaging its required dependency and adding it as
a (conditional) dependency of the curl package.

haampie commented on var/spack/repos/builtin/packages/rtmpdump/package.py:
```diff
@@ -0,0 +1,38 @@
+
+    license("GPL-2.0-or-later")
+
+    version("2021-02-19", commit="f1b83c10d8beb43fcc70a6e88cf4325499f25857")
```
They no longer make releases? I guess it's fine to use the last commit.

tobbez replied:
Yes. Last release was in 2011.

At this point, others packaging rtmpdump seem to be using git snapshots.

haampie commented on var/spack/repos/builtin/packages/rtmpdump/rtmpdump-fix-chunk-size.patch:
```diff
@@ -0,0 +1,48 @@
+https://git.alpinelinux.org/aports/commit/main/rtmpdump/fix-chunk-size.patch?id=bf39fb1177ee77eee6c214a7393cc0054958ce08
+https://git.alpinelinux.org/aports/commit/main/rtmpdump/fix-chunk-size.patch?id=69bc162319b12e9b6c6d3ea345dbf7c218753594
```
I guess it's fine, but out of curiosity: why are those ancient patches not upstreamed?

tobbez replied:
I haven't investigated in detail, but I assume upstream is not responsive, considering these patches are carried by at least two distributions (Alpine and Gentoo).

haampie commented on var/spack/repos/builtin/packages/rtmpdump/package.py:
```diff
@@ -0,0 +1,38 @@
+
+    variant("tls", default="openssl", description="TLS backend", values=("gnutls", "openssl"))
+
+    depends_on("openssl", when="tls=openssl")
```
Maybe put an upperbound on `@:3`? It's using deprecated API, so won't work with an eventual openssl 4.

haampie replied:
```suggestion
    depends_on("openssl@:3", when="tls=openssl")
```

haampie reviewed (commented):
Thanks! Looks good to me, add an upperbound on openssl?

tobbez committed:
rtmpdump: New package

tobbez committed:
curl: Fix librtmp variant

Add the previously missing dependency required for rtmp support.

The variant has been broken since its addition in PR #25166.

Fixes one of the two issues reported in #26887.

tobbez commented:
Upper bound added for openssl.

haampie reviewed (commented):
Thanks, looks excellent!
